apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js

```diff
@@ -0,0 +1,267 @@
+import * as GoogleBlockly from 'blockly/core';
+import msg from '@cdo/locale';
+import {nameComparator} from '@cdo/apps/util/sort';
+import BlockSvgFrame from '../../addons/blockSvgFrame';
+import {createNewDefinitionBlock} from './proceduresBlocks';
+import {convertXmlToJson} from '../../addons/cdoSerializationHelpers';
+import {behaviorDefMutator} from './mutators/behaviorDefMutator';
+import {behaviorGetMutator} from './mutators/behaviorGetMutator';
+
+// In Lab2, the level properties are in Redux, not appOptions. To make this work in Lab2,
+// we would need to send that property from the backend and save it in lab2Redux.
+const useModalFunctionEditor = window.appOptions?.level?.useModalFunctionEditor;
```

Comment:
should we also incorporate the experiment flag here?

Reply:
I'm not sure we need it here until the behavior editor exists. I _think_ the implications are:
1. If the modal function editor is enabled for the level, but the experiment isn't on, the names of the behaviors aren't editable anyway (doesn't seem like a big deal?). Option: We could require both the modal function editor to be enabled and the experiment to be on to keep the names non-editable. 
```
     {
        type: useModalFunctionEditor ? 'field_label' : 'field_input',
        name: 'NAME',
        text: '',
        spellcheck: false,
      },
```
2. If the modal function editor is enabled for a level, we don't render the SVG frame for the behavior definition blocks, even though those definition blocks get defined on the main workspace until the editor works. Option: We could render the SVG frame for the behavior definition blocks even when the modal function editor is enabled as long as the experiment is off. 
```
  if (!useModalFunctionEditor && !this.workspace.noFunctionBlockFrame) {
    this.functionalSvg_ = new BlockSvgFrame(
      this,
      msg.behaviorEditorHeader(),
      'blocklyFunctionalFrame'
    );

    this.setOnChange(function () {
      if (!this.isInFlyout) {
        this.functionalSvg_.render(this.svgGroup_, this.RTL);
      }
    });
  }
```
3. Because there's no alternate callback to open the behavior editor yet, the code below doesn't need to be modified. 
```
  } else if (useModalFunctionEditor) {
    workspace.registerButtonCallback('createNewBehavior', createNewBehavior);
    blockList.push(newBehaviorButton);
```

Thoughts?

Reply:
that all seems fine!